src/back/spv/writer.rs

```diff
@@ -1226,57 +1232,33 @@ impl Writer {
                         let mut vector_id = None;
                         let mut scalar_id = None;
 
-                        match left_inner {
+                        match *left_inner {
                             crate::TypeInner::Vector { size, kind, width } => {
                                 vector_id = Some(left_id);
-                                for (k, v) in self.lookup_type.iter() {
-                                    if let LookupType::Handle(handle) = k {
-                                        let ty = &ir_module.types[*handle];
-                                        match ty.inner {
-                                            crate::TypeInner::Vector {
-                                                size: _size,
-                                                kind: _kind,
-                                                width: _width,
-                                            } => {
-                                                if size == &_size
-                                                    && kind == &_kind
-                                                    && width == &_width
-                                                {
-                                                    result_type_id = Some(*v);
-                                                    break;
-                                                }
-                                            }
-                                            _ => continue,
-                                        }
-                                    }
+                                if let Entry::Occupied(e) =
+                                    self.lookup_type.entry(LookupType::Local(LocalType::Vector {
```

Comment:
we don't have to use the Entry API here, we should just do `if let Some(id) = self.lookup_type.get(...)`